app/models/cohort_lifecycle.rb

```diff
@@ -0,0 +1,366 @@
+      retention_users = active_users.select {| k, _ | retention_user_ids.include?( k ) }
+      retention_users.each_key do | id |
+        user_id = id.to_s.to_sym
+        cohort_data[retention_cohort][user_id]["retention"] = true
+      end
+    end
+  end
+
+  def self.process_interventions( current_day, cohort_data )
+    # intervention 1: no_obs
+    cohort = current_day.to_date.to_s
+    subjects = cohort_data[cohort].select {| _, v | v[:day0] == "no_obs" }
+    subjects.each do | key, value |
+      user = User.where( id: key.to_s.to_i ).first
+      next unless user
+
+      next unless user.locale =~ /en/
+
+      next if user.email.nil? || !user.suspended_at.nil?
+
+      geoip_response = INatAPIService.geoip_lookup( ip: user.last_ip )
+      next unless geoip_response&.results && geoip_response.results.city.present? && geoip_response.results.ll.present?
+
+      geoip_latitude, geoip_longitude = geoip_response.results.ll
+      group = rand( 2 ).zero? ? "A" : "B"
+      value[:observer_appeal_intervention_group] = group
+      next unless group == "A"
+
+      puts "sending...#{user.id}"
+      Emailer.observer_appeal( user, latitude: geoip_latitude, longitude: geoip_longitude ).deliver_now
+    end
+
+    ( 0..6 ).each do | d |
+      cohort = ( current_day - ( d * 24 * 60 * 60 ) ).to_date.to_s
+      slot = "day#{d}".to_s
+      prev_slots = ( 0...d ).map {| q | "day#{q}".to_s }
+
+      puts "#{cohort} #{slot}"
+      next unless cohort_data[cohort]
+
+      # intervention 2: error
+      subjects = cohort_data[cohort].select do | _, v |
+        v[slot] == "error" && prev_slots.all? do | prev_slot |
+          ["no_obs"].include?( v[prev_slot] )
+        end
+      end
+
+      subjects.each do | key, value |
+        user = User.where( id: key.to_s.to_i ).first
+        next unless user
+
+        next unless user.locale =~ /en/
+
+        next if user.email.nil? || !user.suspended_at.nil?
+
+        observation = Observation.where( user_id: user.id, quality_grade: "casual" ).first
+        next unless observation
+
+        error_key = {
+          "georeferenced" => "location",
+          "observed_on" => "date",
+          "recent" => "evidence"
+        }
+        errors = []
+        if !observation.georeferenced? || !observation.observed_on? ||
+            ( !observation.photos? && !observation.sounds? ) || observation.human? ||
+            !observation.quality_metrics_pass?
+
+          missing_fields = ["georeferenced", "observed_on"].reject {| field | observation.public_send( "#{field}?" ) }
+          missing_fields << "evidence" unless observation.photos? || observation.sounds?
+
+          if missing_fields.any?
+            errors.concat( missing_fields.map {| field | error_key.fetch( field, field ) } )
+          elsif ["recent", "evidence", "location", "date"].
+              any? {| field | quality_metric_observation_ids( [observation.id], field ).count == 1 }
```

Comment:
Seems like this method is no longer defined in the context of this code. I see it as an instance method in ObservationAccuracyExperiment. If that's where its singularly defined, maybe that could be turned into a static method, calls updated accordingly in the ObservationAccuracyExperiment model, and then it could be referenced here statically?